File: LeetCode/src/main/java/com/jack/string/LeetCode125.py
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        pattern = r'[^a-z0-9]'
        s = re.sub(pattern, '', s)
        
        left, right = 0, len(s)-1
        while (left< right):
            if(s[left]!= s[right]):
                return False
            
            left+=1
            right-=1
        
        return True

if __name__ == "__main__":
    res = Solution().isPalindrome('A man, a plan, a canal: Panama')
    print(res)

File: LeetCode/src/main/java/com/jack/string/LeetCode125.java
package com.jack.string;

public class LeetCode125 {
    public static void main(String[] args) {

        Solution solution = new Solution();
        boolean res = solution.isPalindrome("A man, a plan, a canal: Panama");

        System.out.println(res);
    }


    static class Solution {
        public boolean isPalindrome(String s) {
            s = s.toLowerCase();
            s = s.replaceAll("[^a-z0-9]", "");

            int len = s.length();
            int left = 0;
            int right = len - 1;

            while (left < right) {
                if (s.charAt(left) != s.charAt(right)) return false;

                left++;
                right--;
            }

            return true;
        }
    }

}
